src/test/optimizer/strategy/between_composition_rule_test.cpp

```diff
@@ -0,0 +1,256 @@
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base_test.hpp"
+#include "gtest/gtest.h"
+
+#include "expression/expression_functional.hpp"
+#include "logical_query_plan/join_node.hpp"
+#include "logical_query_plan/lqp_translator.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/projection_node.hpp"
+#include "logical_query_plan/sort_node.hpp"
+#include "logical_query_plan/stored_table_node.hpp"
+#include "logical_query_plan/union_node.hpp"
+#include "operators/get_table.hpp"
+#include "optimizer/strategy/between_composition_rule.hpp"
+#include "optimizer/strategy/strategy_base_test.hpp"
+#include "statistics/column_statistics.hpp"
+#include "statistics/table_statistics.hpp"
+#include "storage/chunk_encoder.hpp"
+#include "storage/storage_manager.hpp"
+
+#include "utils/assert.hpp"
+
+#include "logical_query_plan/mock_node.hpp"
+
+using namespace opossum::expression_functional;  // NOLINT
+
+namespace opossum {
+
+class BetweenCompositionTest : public StrategyBaseTest {
```

Comment:
Please test that your rule does not
* Pull predicates past nodes that it cannot legally pull past (Aggregates, Non-Inner joins, Union(?))
* Think about what the behaviour should be in diamond-shaped LQPs (Unions) and test accordingly.
* As mentioned elsewhere, test what happens if the right argument to a Predicate is not a Value or the left is not a Column.